src/lib/storage/table.cpp

```diff
@@ -87,5 +88,31 @@ void Table::_append_new_chunk() {
   _chunks.push_back(std::move(new_chunk));
 }
 
-void Table::compress_chunk(ChunkID chunk_id) { throw std::runtime_error("Implement Table::compress_chunk"); }
+void Table::compress_chunk(ChunkID chunk_id) {
+  const auto& uncompressed_chunk = get_chunk(chunk_id);
+  Chunk compressed_chunk = Chunk();
+
+  std::vector<std::future<std::shared_ptr<BaseSegment>>> compressed_segment_futures;
+
+  const auto col_count = column_count();
```

Comment:
Mhm, if we leave this col_count we should maybe also leave dic_size

Reply:
I disagree, just because we can not name this `column_count` does not prevent us from improving the other name :D